src/connector/src/sink/remote.rs

```diff
@@ -228,26 +241,145 @@ async fn validate_remote_sink(param: &SinkParam) -> Result<()> {
+                .as_ref()
+                .ok_or_else(|| anyhow!("Please set delimiter in with option"))?
+                .clone();
+            Box::new(move |row: RowRef<'_>| {
+                let mut keys = vec![];
+                for index in &pk_indices {
+                    keys.push(
+                        row.datum_at(*index)
+                            .ok_or_else(|| anyhow!("No value find in row, index is {}", index))?
+                            .to_text(),
+                    );
+                }
+                Ok(keys.join(&delimiter))
+            })
+        };
+        let json_encoder = JsonEncoder::new_with_es(schema, None);
+        Ok(Self {
+            json_encoder,
+            fn_build_id,
+        })
+    }
+
+    fn convert_chunk(&self, chunk: StreamChunk) -> Result<StreamChunk> {
+        let mut ops = vec![];
+        let mut id_string_builder =
+            <Utf8ArrayBuilder as risingwave_common::array::ArrayBuilder>::new(chunk.capacity());
+        let mut json_builder =
+            <JsonbArrayBuilder as risingwave_common::array::ArrayBuilder>::new(chunk.capacity());
+        for (op, row) in chunk.rows() {
+            ops.push(op);
+            let json = JsonbVal::from(Value::Object(self.json_encoder.encode(row)?));
+            risingwave_common::array::ArrayBuilder::append(
+                &mut id_string_builder,
+                Some(&self.build_id(row)?),
+            );
+            risingwave_common::array::ArrayBuilder::append(
+                &mut json_builder,
+                Some(json.as_scalar_ref()),
+            );
+        }
+        let json_array = risingwave_common::array::ArrayBuilder::finish(json_builder);
+        let id_string_array = risingwave_common::array::ArrayBuilder::finish(id_string_builder);
+        Ok(StreamChunk::new(
+            ops,
+            vec![
+                std::sync::Arc::new(ArrayImpl::Utf8(id_string_array)),
+                std::sync::Arc::new(ArrayImpl::Jsonb(json_array)),
+            ],
+        ))
+    }
+
+    fn build_id(&self, row: RowRef<'_>) -> Result<String> {
+        (self.fn_build_id)(row)
+    }
+}
 pub struct RemoteLogSinker {
     request_sender: BidiStreamSender<JniSinkWriterStreamRequest>,
     response_stream: BidiStreamReceiver<SinkWriterStreamResponse>,
     sink_metrics: SinkMetrics,
+    stream_chunk_converter: StreamChunkConverter,
 }
 
 impl RemoteLogSinker {
-    async fn new(sink_param: SinkParam, writer_param: SinkWriterParam) -> Result<Self> {
+    async fn new(
+        sink_param: SinkParam,
+        writer_param: SinkWriterParam,
+        sink_name: &str,
+    ) -> Result<Self> {
         let SinkWriterStreamHandle {
             request_sender,
             response_stream,
         } = EmbeddedConnectorClient::new()?
-            .start_sink_writer_stream(sink_param, SinkPayloadFormat::StreamChunk)
+            .start_sink_writer_stream(&sink_param, SinkPayloadFormat::StreamChunk, sink_name)
```

Comment:
In both `start_sink_writer_stream` methods, we'd better pass the separate payload schema as a new parameter to the methods instead of choosing different schema for different sink name in the implementation of the `start_sink_writer_stream` methods.